sweep/fee_bumper.go

```diff
@@ -84,6 +84,11 @@ const (
 	// TxConfirmed is sent when the tx is confirmed.
 	TxConfirmed
 
+	// TxFatal is sent when the inputs in this tx cannot be retried. Txns
+	// will end up in this state if they have encountered a non-fee related
+	// error, which means they cannot be retried with increased budget.
+	TxFatal
```

Comment:
wonder if we should rename `TxFailed` to something like `TxFeeFailed` or `TxInputsFailed` or `TxFailedRetriable` or something to make it clear what the difference between that and this is.

Reply:
also just checking my understanding: we mark a tx as this and hence _all its inputs_ as terminal meaning we wont retry any of those inputs even if they have a deadline? 

should we not re-submit those inputs so they are regrouped (or potentially handled individually)? or is that the plan in future ? just wondering if one bad input will result in us dropping a bunch of inputs

Reply:
agree the long term goal should be to make sure we know the input which is causing the problems.

Reply:
> we mark a tx as this and hence all its inputs as terminal meaning we wont retry any of those inputs even if they have a deadline?

yeah correct, so this one bad input will affect all other inputs, which is not great. There's a TODO for that (the 6th item out of ten in #8680, and we are at one🤦🏻), the idea is to use `testmempoolaccept` to do a binary search on which input is causing the failure then kick it out - but i guess we need to think about how to handle it in neutrino as there's no mempool.